### ISSUE TITLE
Error passing ** keyword args to function with multiple optional arguments of different types

### ISSUE THREAD

**Issue by pdebelak** [python/mypy#7553]
This is a bug report present in mypy 0.720 (currently lastest mypy).

I get an unexpected type error when trying to pass `**` keyword args to a function with multiple optional arguments of different types.

Take this example code:

```python
from typing import Optional


def a_func(arg1: Optional[str] = None, arg2: Optional[int] = None):
    pass


a_func(**{"arg2": 1})
```

I would expect this to report no type errors, but instead I get:

```
error: Argument 1 to "a_func" has incompatible type "**Dict[str, int]"; expected "Optional[str]"
```

Is there a way to get this to pass mypy checking? My real code involves passing a dynamic kwarg (like `year`, `month`, or `week`) along with a length to create several different`dateutil.relativedelta` instances so rewriting it to not use `**` would involve rewriting the real code to have a large number of conditionals.

**Comment by ilevkivskyi** [comment]
This is a duplicate of https://github.com/python/mypy/issues/5382
